addon/mixins/base-query-manager.js

```diff
@@ -0,0 +1,24 @@
+import Ember from 'ember';
+import QueryManager from 'ember-apollo-client/apollo/query-manager';
+
+const { getOwner, Mixin } = Ember;
+
+export default Mixin.create({
+  apollo: null,
+
+  init() {
+    this._super(...arguments);
+    let owner = getOwner(this);
+
+    // I think this is necessary in order to make this register properly when used in
+    // unit or integration tests, but I'm not sure it's the right way to solve it.
+    if (!owner.hasRegistration('object:apollo-query-manager')) {
+      owner.register('object:apollo-query-manager', QueryManager, {
+        singleton: false,
+      });
+    }
```

Comment:
Rather than having the query managers be container-managed, one alternative might be to introduce a simple`createQueryManager` method to the existing `apollo` service:

```js
createQueryManager() {
  return QueryManager.create({ apollo: this });
}
```

This base mixin could then just rely on the service and instantiate a query manager on init as it's already doing, but you wouldn't have to deal with the custom registration.

Integration tests should just work, and unit tests could continue to just have `needs: ['service:apollo']`.